scripts/west/zap_gui.py

```diff
@@ -31,33 +33,52 @@ def do_add_parser(self, parser_adder):
                             default=DEFAULT_MATTER_PATH, help='Path to Matter SDK')
+        parser.add_argument('--clusters', nargs='+',
+                            help="Paths to the XML files that contain the external cluster definitions")
+        parser.add_argument('-c', '--clear', action='store_true', help='Clear the ZAP cache')
         return parser
 
     def do_run(self, args, unknown_args):
-        if args.zap_file:
-            zap_file_path = args.zap_file
-        else:
-            zap_file_path = find_zap()
+        zap_file_path = args.zap_file or find_zap()
+        zcl_json_path = Path(args.zcl_json).absolute() if args.zcl_json else DEFAULT_ZCL_PATH
 
-        if args.zcl_json:
-            zcl_json_path = args.zcl_json.absolute()
-        else:
-            zcl_json_path = args.matter_path / 'src/app/zap-templates/zcl/zcl.json'
+        if args.clusters:
+            # If the user provided the clusters and the zcl.json file provided by -j argument does not exist
+            # we will create a new zcl.json file according to the base zcl.json file in DEFAULT_ZCL_PATH.
+            # If the provided zcl.json file exists, we will use it as a base and update with a new cluster.
+            base_zcl = zcl_json_path if zcl_json_path.exists() else DEFAULT_ZCL_PATH
+            add_cluster_to_zcl(base_zcl, args.clusters, zcl_json_path)
+        elif not zcl_json_path.exists():
+            # If clusters are not provided, but user provided a zcl.json file we need to check whether the file exists.
+            print(f"ZCL file not found: {zcl_json_path}")
+            return
 
-        app_templates_path = args.matter_path / 'src/app/zap-templates/app-templates.json'
+        app_templates_path = DEFAULT_APP_TEMPLATES_PATH
+
+        print(f"Using ZAP file: {zap_file_path}")
+        print(f"Using ZCL file: {zcl_json_path}")
+        print(f"Using app templates: {app_templates_path}")
 
         zap_installer = ZapInstaller(args.matter_path)
         zap_installer.update_zap_if_needed()
-        zap_cache_path = zap_installer.get_install_path() / ".zap"
+
+        # The zcl.json path in the .zap file must be the same as the one provided by the user
+        # If not, update the .zap file with the new relative path to the zcl.json file.
+        # After that we must clear the ZAP cache.
+        if not check_zcl_in_zap(zap_file_path, zcl_json_path):
+            update_zcl_in_zap(zap_file_path, zcl_json_path)
+            args.clear = True
 
         cmd = [zap_installer.get_zap_path()]
         cmd += [zap_file_path] if zap_file_path else []
         cmd += ["--zcl", zcl_json_path]
         cmd += ["--gen", app_templates_path]
-        cmd += ["--stateDirectory", zap_cache_path]
-
+        cmd += ["--stateDirectory", zap_installer.get_cache_path()]
         self.check_call([str(x) for x in cmd])
+
+        # clear cache after the usage
+        zap_installer.clear_cache()
```

Comment:
Shouldn't we check `if args.clear` before that?

Reply:
I forgot to remove args.clear :) Now cache is always temporary.